File: demo/pdf/generate-pdf.js
const puppeteer = require('puppeteer');
const path = require('path');
const fs = require('fs');

async function generatePDF() {
    console.log('Starting PDF generation...');
    
    try {
        // Launch headless browser
        const browser = await puppeteer.launch({
            headless: true,
            args: ['--no-sandbox', '--disable-setuid-sandbox']
        });
        
        const page = await browser.newPage();
        
        // Read the HTML file
        const htmlPath = path.join(__dirname, '1.html');
        const htmlContent = fs.readFileSync(htmlPath, 'utf8');
        
        // Set the content
        await page.setContent(htmlContent, {
            waitUntil: 'networkidle0'
        });
        
        // Wait for fonts to load
        await page.evaluateHandle('document.fonts.ready');
        
        // Generate PDF with print-friendly settings
        await page.pdf({
            path: path.join(__dirname, 'vcpulse-weekly-brief.pdf'),
            format: 'A4',
            printBackground: true,
            margin: {
                top: '0mm',
                right: '0mm',
                bottom: '0mm',
                left: '0mm'
            },
            displayHeaderFooter: false,
            preferCSSPageSize: true
        });
        
        console.log('PDF generated successfully: vcpulse-weekly-brief.pdf');
        
        // Also generate a US Letter version
        await page.pdf({
            path: path.join(__dirname, 'vcpulse-weekly-brief-letter.pdf'),
            format: 'Letter',
            printBackground: true,
            margin: {
                top: '0mm',
                right: '0mm',
                bottom: '0mm',
                left: '0mm'
            },
            displayHeaderFooter: false,
            preferCSSPageSize: true
        });
        
        console.log('PDF generated successfully: vcpulse-weekly-brief-letter.pdf (US Letter)');
        
        await browser.close();
        
    } catch (error) {
        console.error('Error generating PDF:', error);
        process.exit(1);
    }
}

// Run the PDF generation
generatePDF();

File: demo/pdf/generate-pdf.py
#!/usr/bin/env python3
"""
Generate PDF from VCPulse Weekly Intelligence Brief HTML
Requires: pip install weasyprint
"""

import os
import sys
from pathlib import Path

try:
    from weasyprint import HTML, CSS
except ImportError:
    print("Error: weasyprint not installed.")
    print("Please install it with: pip install weasyprint")
    sys.exit(1)

def generate_pdf():
    """Generate PDF from HTML file"""
    # Get the directory of this script
    script_dir = Path(__file__).parent
    
    # Input and output paths
    html_file = script_dir / "1.html"
    pdf_file = script_dir / "vcpulse-weekly-brief.pdf"
    
    if not html_file.exists():
        print(f"Error: HTML file not found: {html_file}")
        sys.exit(1)
    
    print("Starting PDF generation...")
    
    try:
        # Generate PDF
        HTML(filename=str(html_file)).write_pdf(
            str(pdf_file),
            stylesheets=[CSS(string='''
                @page {
                    size: A4;
                    margin: 0;
                }
                body {
                    margin: 0;
                }
            ''')]
        )
        
        print(f"PDF generated successfully: {pdf_file}")
        
        # Also generate US Letter version
        pdf_file_letter = script_dir / "vcpulse-weekly-brief-letter.pdf"
        HTML(filename=str(html_file)).write_pdf(
            str(pdf_file_letter),
            stylesheets=[CSS(string='''
                @page {
                    size: Letter;
                    margin: 0;
                }
                body {
                    margin: 0;
                }
            ''')]
        )
        
        print(f"PDF generated successfully: {pdf_file_letter} (US Letter)")
        
    except Exception as e:
        print(f"Error generating PDF: {e}")
        sys.exit(1)

if __name__ == "__main__":
    generate_pdf()